### PR TITLE
Issue #256: trim back old meta data files to rely on checkstyle

### PR DESCRIPTION
**Pull request by rnveach**
Issue #256

This is the first of the attempts to remove the old meta data files. This based on comment at https://github.com/checkstyle/eclipse-cs/issues/256#issuecomment-907832411 .

Currently I cannot remove the XML and property files completely. Doing so either causes exceptions or causes the checks pulled from checkstyle to disappear.

When this is merged, https://github.com/checkstyle/eclipse-cs/issues/113 can be closed as it no longer applies.

Issue #256 can only be closed when the XML/property files are completely gone showing we are completely reliant on Checkstyle.

### CODE REVIEW

**Comment by rnveach** [comment]
As I am new to plugin maintenance, what is required for this to be fully vetted and merged. This is a good step forward to making eclipse-cs updates easier.

**Comment by rnveach** [comment]
> `<quickfix classname="net.sf.eclipsecs.ui.quickfixes.misc.ArrayTypeStyleQuickfix"/>`

When reviewing the changes, I noticed this line.
Without this link in the XML file, does this mean quickfixes will be broken?

File: net.sf.eclipsecs.checkstyle/metadata/com/puppycrawl/tools/checkstyle/checks/blocks/checkstyle-metadata.xml [hunk 1]
```diff
@@ -4,165 +4,5 @@
...
 <checkstyle-metadata>
     <rule-group-metadata name="%Blocks.group" priority="800">
-
-        <rule-metadata name="%AvoidNestedBlocks.name" internal-name="AvoidNestedBlocks" parent="TreeWalker">
-            <alternative-name internal-name="com.puppycrawl.tools.checkstyle.checks.blocks.AvoidNestedBlocksCheck"/>
-            <description>%AvoidNestedBlocks.desc</description>
-            <property-metadata name="allowInSwitchCase" datatype="Boolean" default-value="false">
-                <description>%AvoidNestedBlocks.allowInSwitchCase</description>
-            </property-metadata>
-            <quickfix classname="net.sf.eclipsecs.ui.quickfixes.blocks.AvoidNestedBlocksQuickfix"/>
```
Comment (rnveach) [inline net.sf.eclipsecs.checkstyle/metadata/com/puppycrawl/tools/checkstyle/checks/blocks/checkstyle-metadata.xml:14]: When reviewing the changes, I noticed this line.
Without this link in the XML file, does this mean quickfixes will be broken?

Comment (rnveach) [reply net.sf.eclipsecs.checkstyle/metadata/com/puppycrawl/tools/checkstyle/checks/blocks/checkstyle-metadata.xml:14]: Break goes beyond this change and release. See https://github.com/checkstyle/eclipse-cs/issues/315 .

**Comment by Calixte** [comment]
> As I am new to plugin maintenance, what is required for this to be fully vetted and merged?

I usually make sure `mvn clean test package` passes, then run the new plugin, create a new configuration, add some checks to it, create a file, write some code that fails those checks, verify I get warnings, and while I do that, look for anything weird. Not very automated.

**Review by Calixte — APPROVED** [review]

**Comment by rnveach** [comment]
> mvn clean test package passes, then run the new plugin, create a new configuration, add some checks to it, create a file, write some code that fails those checks, verify I get warnings

I have done this now. I created and use my own config, I have switched between configs. I also played with configs between Workspace (Preferences) and Project (Properties).

Is that the only 2 places where eclipse-cs has it's own dialogs, Workspace (Preferences) and Project (Properties)?

**Comment by Calixte** [comment]
Sounds good to me. Do you have permission to merge?

**Comment by rnveach** [comment]
I do, I just normally don't do it for my own PRs so someone else can give the finalize.

If you are not aware, I am also an admin of checkstyle like @romani. I have worked there and sevntu for quiet a long time, but have not really been involved in eclipse-cs.

**Comment by Calixte** [comment]
I wasn't sure if you were missing permissions on this particular repo, or if you wanted someone else to merge.
I'll do it.

**Comment by rnveach** [comment]
No need to go back and forth on this. Feel free to merge if you don't want any other reviews or need anything else from me.

**Pull request merged**